### PR TITLE
opentelemetry: export with_examples parameter

### PR DESCRIPTION
**Pull request by jiaoew1991**
Specify library name and version:  **opentelemetry/all**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 1]
```diff
@@ -61,6 +62,7 @@ class OpenTelemetryCppConan(ConanFile):
         "with_logs_preview": False,
         "with_async_export_preview": False,
         "with_metrics_exemplar_preview": False,
+        "with_examples": True,
```
Comment (jcar87) [inline recipes/opentelemetry-cpp/all/conanfile.py:65]: hi @jiaoew1991  - thanks for opening this PR. 

We typically exclude building examples, most recipes where the underlying build scripts build examples by default would have them explicitly disabled.  The rationale is that they are _not_ part of the public API of the library and thus not necessary for consumers, and the Conan test package serves as a minimal example to cover this - Is it not the case with `opentelemetry-cpp`?

Trying to understand the use

Comment (jiaoew1991) [reply recipes/opentelemetry-cpp/all/conanfile.py:65]: I want to disable the examples too, but in the original recipe the example was not disabled and the cmake of  `opentelemetry-cpp` build the examples by default. I do not want to break the compatibility so keep building the example 😅

**Comment by jiaoew1991** [comment]
Hi @jcar87 , it seems the CI env still not ready 😭
image

**Comment by rbharvs** [comment]
Hi all,

I opened a Github https://github.com/conan-io/conan-center-index/issues/16296 recently related to an issue I discovered that was introduced to the package recipe recently. Namely, a change was made that to the recipe that caused Conan to build the example files in the OpenTelemetry repo. Unfortunately, there is a mistake in one of their example files that prevents it from building in C++ 20.

It seems like this PR would at least expose the ability to turn off the building of examples. However, it might be best to just set `tc.variables["WITH_EXAMPLES"] = False` so that it's disabled entirely.

What will it take to get this PR over the line? (I'm unfamiliar with contributing to this repo)

Thank you!

cc @jcar87 @jiaoew1991

**Pull request closed without merging**